wgsl/index.bs

```diff
@@ -896,33 +900,143 @@ A <dfn noexport>type rule</dfn> has two parts:
     * How the expression is used in a statement.
     * Optionally, other static context.
 
-Each distinct type parameterization for a type rule is called an <dfn noexport>overload</dfn>.
-For example, [[#arithmetic-expr|unary negation]] (an expression of the form `-`|e|)
-has eight overloads, because its type rules are parameterized by a type |T| that can be any of:
-* [=i32=]
-* vec2&lt;i32&gt;
-* vec3&lt;i32&gt;
-* vec4&lt;i32&gt;
-* [=f32=]
-* vec2&lt;f32&gt;
-* vec3&lt;f32&gt;
-* vec4&lt;f32&gt;
+Type rules may have <dfn noexport>type parameters</dfn> in their preconditions and conclusions.
```

Comment:
This is a really nice passage. We should find a way to land this independent of the inferencing aspect of the PR.

Reply:
I'm suggesting splitting out lines 903..965.  I think that stands on its own and is a great improvement the spec as-is.

Reply:
I've extracted this as #2682.